### PR TITLE
fixes medbots

### LINKED ISSUE #11901: Special (green, blue, red) medical bots revet to standard (white) medkits when destroyed
(WEB REPORT BY: sonixapache REMOTE: 172.93.109.202:7777)

> Revision (Should be above if you're viewing this from ingame!)
> d5e11d4f8fb61ecf01b55e2103ba73471e7a22c1
> 
> General description of the issue
> Roboticist turned a green medkit into a green medbot.
> Green medbot was then destroyed.
> Out of the destroyed remains came a white medkit.

### LINKED ISSUE #12653: Medibots do not check for overdose directly before injecting.
#### Description of issue
Raised in #12262.
Medibots will check if the patient has meds to begin with, but will not check just before injecting, leading to overdoses.

#### Difference between expected and actual behavior

Medibot stops itself from taking a life.
Medibot takes a life.

#### Steps to reproduce 

Stand near a medibot with brute damage, have it start injecting you, eat 29 units of bicard in a pill, get OD'd on by the bot when it finishes injecting.

### LINKED ISSUE #13587: medibots are broken
(WEB REPORT BY: pvak REMOTE: 172.93.109.202:7777)
>Revision (Should be above if you're viewing this from ingame!)

>General description of the issue
they have been broken for a while maybe the pai medibot pr did it.
medibots now spam injections and when i used a mass spectometer on the dude injected he had nothing in the blood 

>Steps to reproduce if possible
be hurt near a medibot
get 9999 injection messages
no meds in the blood

### PR DESCRIPTION
**Pull request by ghost**
Removes a shit ton of src. for readability
fixes #11901
fixes #13587 
fixes #12653
fixes medbots not dropping their associated medkit
fixes medbots not injecting shit/not stopping injecting shit
fixes a slurry of medbots potentially overdosing a patient by all injecting them with the same stuff

:cl:
* rscadd: Medbots now drop their associated medkit when destroyed
* rscadd: Medbots no longer OD the patient if a large amount try to inject one person at once

### CODE REVIEW

**Comment by JustSumBody** [comment]
Removing src is a nonstandard formatting change(even though it's the way it should be)

**Comment by ghost** [comment]
Everything was src.

It was as if celt got into an argument with cluster over src so src.'d everything

**Comment by gbasood** [comment]
@JustSumBody `src.` removal is
pretty goddamn standard
even if it isn't written down anywhere, `src.` should really mostly be used if it improves readability

**Comment by Exxion** [comment]
Removing it indiscriminately like this hinders PR review though
And no, it isn't standard. It was a fad for a few days until people realized how stupid it was to make such a fuss about it

File: code/game/machinery/bots/medbot.dm [hunk 1]
```diff
@@ -472,47 +472,54 @@
...
-		visible_message("<span class='danger'>[src] injects [src.patient] with the syringe!</span>")
+			patient.reagents.add_reagent(reagent_id,injection_amount)
+
+		visible_message("<span class='danger'>[src] injects [patient] with the syringe!</span>")
 
-	src.icon_state = "[src.icon_initial][src.on]"
-	src.currently_healing = 0
+	icon_state = "[icon_initial][on]"
+	currently_healing = 0
+	reagent_id = null
```
Comment (ghost) [inline code/game/machinery/bots/medbot.dm]: Primary change one - Upon injection, the reagent_id is set to null, which forces the medbot to re-evaluate the patient, which in turn leads to them not constantly injecting nothing

File: code/game/machinery/bots/medbot.dm [hunk 2]
```diff
@@ -472,47 +472,54 @@
...
 	..()
 
 /obj/machinery/bot/medbot/explode()
-	src.on = 0
+	on = 0
 	visible_message("<span class='danger'>[src] blows apart!</span>", 1)
 	var/turf/Tsec = get_turf(src)
 
-	new /obj/item/weapon/storage/firstaid(Tsec)
+	switch(skin)
```
Comment (ghost) [inline code/game/machinery/bots/medbot.dm:511]: Primary change two - Upon medbot destruction, it switches through skins much like it does on new, where it then drops the corresponding medkit upon destruction, defaulting to the basic medkit

**Comment by ghost** [comment]
There were 246 instances of src., of which 0 were necessary

**Comment by Exxion** [comment]
Yeah, and I don't really care whether or not they're left there, but removing them all at once in a PR whose functional part only touches very few of them makes review harder

Thanks for the line comment pointing out the functional part, though.

**Comment by CptWad** [comment]
While you are at it, do medbots still check if they are overdosing people? I have had them overdose me on multiple occasions

**Comment by ghost** [comment]
This also fixes #13587

**Comment by ghost** [comment]
The OD check Ill redo, it currently checks on patient scan rather than on inject

**Comment by ghost** [comment]
Done. Now if there's a clusterfuck of medbots, only one will inject you and the rest'll just fail the injection

File: code/game/machinery/bots/medbot.dm [hunk 3]
```diff
@@ -472,47 +472,58 @@
...
+			reagent_glass.reagents.trans_to(patient,injection_amount) //Inject from beaker instead.
+			reagent_glass.reagents.reaction(patient, 2)
+			succesful_inject = 1
 		else
-			src.patient.reagents.add_reagent(reagent_id,src.injection_amount)
-		visible_message("<span class='danger'>[src] injects [src.patient] with the syringe!</span>")
-
-	src.icon_state = "[src.icon_initial][src.on]"
-	src.currently_healing = 0
+			if(!patient.reagents.has_reagent(reagent_id) && !emagged) //Somebody got there first
```
Comment (ghost) [inline code/game/machinery/bots/medbot.dm:483]: Primary change three - Now it double checks if the patient has the reagent it's going to inject into said patient. First when evaluating, and secondly just before injection.

If the patient does have the chemical, it does not return a succesful injection, and therefore does not print that the patient was injected with the syringe

**Comment by SonixApache** [comment]
yaaay

**Pull request merged**